### PR TITLE
working version for automl od/seg convertor

### PR DESCRIPTION
**Pull request by yiyoulin**
**Object Detection**
[Pipeline](https://ml.azure.com/experiments/id/d6821871-54ce-45eb-868c-0128178d1993/runs/6eee888e-8cda-4cf3-98b6-6898856039c9?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/automlimage_eastus2_rg/providers/Microsoft.MachineLearningServices/workspaces/risha-ws-eastus2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
[Model](https://ml.azure.com/registries/automl-images-dev/models/AutoML-Image-Object-Detection/version/2?tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview)

**Segmentation**
[Pipeline](https://ml.azure.com/experiments/id/61aaf580-4c5a-469d-8f89-70b6fd6c5ae8/runs/08be1aad-9d24-47f3-867d-0b449bae995e?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/automlimage_eastus2_rg/providers/Microsoft.MachineLearningServices/workspaces/risha-ws-eastus2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
[Model](https://ml.azure.com/registries/automl-images-dev/models/AutoML-Image-Instance-Segmentation/version/2?tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview)

### CODE REVIEW

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/config.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ class SupportedTasks(_CustomEnum):
     # AutoML Tasks
     IMAGE_CLASSIFICATION = "image-classification"
     IMAGE_CLASSIFICATION_MULTILABEL = "image-classification-multilabel"
+    AUTOML_OBJECT_DETECTION = "automl-object-detection"
+    AUTOML_INSTANCE_SEGMENTATION = "automl-instance-segmentation"
```
Comment (sharma-riti) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/config.py:54]:
```suggestion
    AUTOML_OBJECT_DETECTION = "image-object-detection"
    AUTOML_INSTANCE_SEGMENTATION = "image-instance-segmentation"
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/config.py [hunk 2]
```diff
@@ -50,6 +50,8 @@ class SupportedTasks(_CustomEnum):
     # AutoML Tasks
     IMAGE_CLASSIFICATION = "image-classification"
     IMAGE_CLASSIFICATION_MULTILABEL = "image-classification-multilabel"
+    AUTOML_OBJECT_DETECTION = "automl-object-detection"
```
Comment (sharma-riti) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/config.py:53]: or rename "MM_OBJECT_DETECTION" to "IMAGE_OBJECT_DETECTION"?

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py [hunk 3]
```diff
@@ -761,7 +761,9 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
         if self._task not in [
             SupportedTasks.IMAGE_CLASSIFICATION.value,
-            SupportedTasks.IMAGE_CLASSIFICATION_MULTILABEL.value
+            SupportedTasks.IMAGE_CLASSIFICATION_MULTILABEL.value,
+            SupportedTasks.AUTOML_OBJECT_DETECTION.value,
+            SupportedTasks.AUTOML_INSTANCE_SEGMENTATION.value,
```
Comment (sharma-riti) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:766]:
```suggestion
            SupportedTasks.IMAGE_OBJECT_DETECTION.value,
            SupportedTasks.IMAGE_INSTANCE_SEGMENTATION.value,
```

**Review by sharma-riti — APPROVED** [review]

**Comment by sharma-riti** [comment]
can you add model card details as well?

**Review by sharma-riti — COMMENTED** [review]
can you add model card details as well?

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/automl/od_is_classes.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+COCO_CLASSES = ['person', 'bicycle', 'car', 'motorcycle', 'airplane',
```
Comment (sharma-riti) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/automl/od_is_classes.py:1]: I think you would need a copyright header... maybe convert it to text file?

**Review by sharma-riti — APPROVED** [review]

**Review by sharma-riti — APPROVED** [review]

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py [hunk 5]
```diff
@@ -836,18 +935,20 @@ def save_as_mlflow(self):
...
         from azureml.automl.dnn.vision.common.mlflow.mlflow_model_wrapper import MLFlowImagesModelWrapper
         from azureml.automl.dnn.vision.common.model_export_utils import _get_scoring_method
-
-        mlflow_model_wrapper = MLFlowImagesModelWrapper(
-            model_settings={},
-            task_type=self._task,
-            scoring_method=_get_scoring_method(self._task),
-        )
+        is_yolo = self._task == SupportedTasks.IMAGE_OBJECT_DETECTION.value
+        mlflow_model_wrapper = MLFlowImagesModelWrapper(model_settings={},
```
Comment (vivek-dani) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:939]: We have some json file in most of the model cards, which is read and then used to do logging. This is then used to track usage of this model. Madhu worked on it. Check once with Madhu if logging would happen when this model card is used.

**Review by vivek-dani — DISMISSED** [review]

File: assets/models/system/automl_instance_segmentation/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    Standard_D64as_v4, Standard_D96a_v4, Standard_D96as_v4, Standard_FX4mds, Standard_F8s_v2,
+    Standard_FX12mds, Standard_F16s_v2, Standard_F32s_v2, Standard_F48s_v2, Standard_F64s_v2,
+    Standard_F72s_v2, Standard_FX24mds, Standard_FX36mds, Standard_FX48mds, Standard_E4s_v3,
+    Standard_E8s_v3, Standard_E16s_v3, Standard_E32s_v3, Standard_E48s_v3, Standard_E64s_v3,
+    Standard_NC4as_T4_v3, Standard_NC6s_v3, Standard_NC8as_T4_v3, Standard_NC12s_v3,
+    Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3, Standard_NC24ads_A100_v4,
+    Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4,
+    Standard_ND40rs_v2
+tags:
+  license: apache-2.0
```
Comment (PhaniShekhar) [inline assets/models/system/automl_instance_segmentation/spec.yaml:22]: Is this license information correct. I see that we have it listed under different license in our azuremlcli repo is different: https://msdata.visualstudio.com/Vienna/_git/AzureMlCli?path=/src/azureml-automl-dnn-vision/azureml/automl/dnn/vision/common/NOTICE&version=GBmaster&line=734&lineEnd=735&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

File: assets/models/system/automl_object_detection/description.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+For more information, see [how to configure experiments](https://learn.microsoft.com/azure/machine-learning/how-to-auto-train-image-models?view=azureml-api-2&tabs=python#configure-experiments)
+
+### Code samples
+
+Task|Use case|Dataset|Python sample (Notebook)|CLI with YAML
+|---|--|--|--|--|
+Image object detection|Image object detection|[fridgeObjects](https://cvbp-secondary.z19.web.core.windows.net/datasets/object_detection/odFridgeObjects.zip)|<a href="https://github.com/Azure/azureml-examples/blob/main/sdk/python/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb" target="_blank">[fridgeobjects-object-detection.ipynb](https://github.com/Azure/azureml-examples/blob/main/sdk/python/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb)</a>|<a href="https://github.com/Azure/azureml-examples/tree/sdk-preview/cli/jobs/automl-standalone-jobs/cli-automl-image-object-detection-task-fridge-items" target="_blank">cli-automl-image-object-detection-task-fridge-items.yml</a>
+
+
+### Sample inputs and outputs
```
Comment (PhaniShekhar) [inline assets/models/system/automl_object_detection/description.md:31]: Nit: (for real time inference)

File: assets/models/system/automl_object_detection/spec.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    Standard_D64as_v4, Standard_D96a_v4, Standard_D96as_v4, Standard_FX4mds, Standard_F8s_v2,
+    Standard_FX12mds, Standard_F16s_v2, Standard_F32s_v2, Standard_F48s_v2, Standard_F64s_v2,
+    Standard_F72s_v2, Standard_FX24mds, Standard_FX36mds, Standard_FX48mds, Standard_E4s_v3,
+    Standard_E8s_v3, Standard_E16s_v3, Standard_E32s_v3, Standard_E48s_v3, Standard_E64s_v3,
+    Standard_NC4as_T4_v3, Standard_NC6s_v3, Standard_NC8as_T4_v3, Standard_NC12s_v3,
+    Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3, Standard_NC24ads_A100_v4,
+    Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4,
+    Standard_ND40rs_v2
+tags:
+  license: mit
```
Comment (PhaniShekhar) [inline assets/models/system/automl_object_detection/spec.yaml:22]: Please check this license as well.

**Review by PhaniShekhar — CHANGES_REQUESTED** [review]

File: assets/models/system/automl_object_detection/spec.yaml (hunk 8 above)
Comment (sharma-riti) [reply assets/models/system/automl_object_detection/spec.yaml:22]: I looked at this one - https://github.com/ultralytics/yolov5/blob/master/LICENSE

File: assets/models/system/automl_instance_segmentation/spec.yaml (hunk 6 above)
Comment (sharma-riti) [reply assets/models/system/automl_instance_segmentation/spec.yaml:22]: https://arxiv.org/abs/1703.06870 --> https://github.com/facebookresearch/Detectron/blob/main/LICENSE

**Review by PhaniShekhar — APPROVED** [review]

**Review by sharma-riti — APPROVED** [review]

**Pull request merged**